test/integration/slack_test.exs

```diff
@@ -61,4 +61,58 @@ defmodule Integration.SlackTest do
     message = send_message("@#{@bot}: operable:echo blah", private_channel)
     assert_response "blah", [after: message], private_channel
   end
+
+  test "redirecting from a private channel", %{user: user} do
+    user |> with_permission("operable:echo")
+    private_channel = "group_ci_bot_testing"
+    marker = send_message("Redirect test starts here")
+
+    send_message("@#{@bot}: operable:echo blah > #ci_bot_testing", private_channel)
+    assert_response "blah", [after: marker]
+  end
+
+  test "redirecting to a private channel", %{user: user} do
+    user |> with_permission("operable:echo")
+    private_channel = "group_ci_bot_testing"
+    marker = send_message("Redirect test starts here", private_channel)
+
+    send_message("@#{@bot}: operable:echo blah > ##{private_channel}")
+    assert_response "blah", [after: marker], private_channel
+  end
+
+  test "redirecting to 'here'", %{user: user} do
+    user |> with_permission("operable:echo")
+
+    message = send_message("@#{@bot}: operable:echo blah > here")
+    assert_response "blah", [after: message]
+  end
+
+  test "redirecting to 'me'", %{user: user} do
+    user |> with_permission("operable:echo")
+    marker = send_message("echo here", "@#{@user}")
+
+    send_message("@#{@bot}: operable:echo blah > me")
+    # Since Cog responds when direct messaging it we have to assert
+    # that both our marker text and message test exist.
+    assert_response "here\nblah", [after: marker, count: 2], "@#{@user}"
+  end
+
+  test "redirecting to a specific user", %{user: user} do
+    user |> with_permission("operable:echo")
+    marker = send_message("echo here", "@#{@user}")
+
+    send_message("@#{@bot}: operable:echo blah > @#{@user}")
+    # Since Cog responds when direct messaging it we have to assert
+    # that both our marker text and message test exist.
+    assert_response "here\nblah", [after: marker, count: 2], "@#{@user}"
+  end
+
+  test "redirecting to another channel", %{user: user} do
+    user |> with_permission("operable:echo")
+    redirect_channel = "ci_bot_redirect_tests"
+    marker = send_message("echo here", redirect_channel)
+
+    send_message("@#{@bot}: operable:echo blah > ##{redirect_channel}")
+    assert_response "blah", [after: marker], redirect_channel
```

Comment:
Why are you not asserting a response of `"here\nblah"` for this test, but you are for the `"redirecting to a specific user"`. They seem to be structurally the same.